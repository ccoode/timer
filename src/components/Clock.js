import React, { PropTypes } from 'react'

/**
 * timeout(ms) => MM:ss 
 */
function Clock({ timeout }) {
    return (
        <section className="clock">
            {convertMS(timeout)}
        </section>
    )
}

Clock.propTypes = {
    timeout: PropTypes.number
}



function convertMS(ms) {
    if (ms >= 0) {
        var s = ms / 1000,
            sec = Math.ceil(s % 60),
            min = Math.floor(s / 60)
        if (sec === 60) {
            min += 1
            sec = 0
        }
        return pad(min) + ":" + pad(sec)
    }
}

function pad(number) {
    if (number < 10) {
        return '0' + number
    }
    return number
}

export default Clock