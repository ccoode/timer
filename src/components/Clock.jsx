import React, { PropTypes } from 'react'

function pad(number) {
  if (number < 10) {
    return `0${number}`
  }
  return number
}

function convertMS(ms) {
  if (ms < 0) return 0
  const s = ms / 1000
  let sec = Math.ceil(s % 60)
  let min = Math.floor(s / 60)
  if (sec === 60) {
    min += 1
    sec = 0
  }
  return `${pad(min)}:${pad(sec)}`
}

/**
 * timeout(ms) => MM:ss
 */
function Clock({ timeout }) {
  return (
    <section className="clock">
      {convertMS(timeout)}
    </section>
  )
}

Clock.propTypes = {
  timeout: PropTypes.number.isRequired,
}

export default Clock
