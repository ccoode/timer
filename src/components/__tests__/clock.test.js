import Clock from '../Clock'
import { h } from 'preact'
import { expect } from 'chai'

test('clock', () => {
  expect(<Clock timeout={2000} />).to.deep.equal(
    <section className='clock'>00:02</section>
  )
})
