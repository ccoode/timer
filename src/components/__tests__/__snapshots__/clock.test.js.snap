// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clock 1`] = `
<section
  className="clock"
>
  00:02
</section>
`;
